### DISCUSSION TITLE
Controller now pressing random inputs on its own (both wired and wireless)

### DISCUSSION THREAD

**Discussion by ZxSol** [nefarius/DsHidMini#170] [Support]
I have been using DSHIDMINI for almost a year without any issues, until today, when I was playing Blazblue Central Fiction, I was having a difficult time navigating the ingame menu for some reason, when I went to training mode, I found out my controller has started rapidly inputting buttons on its own(the game has an input history shown in the left side of the attached screenshot that lets you see the last buttons pressed in a sequence)

I thought it was maybe a problem with my usb cord, so i closed the game, unplugged and used wireless bluetooth connection, then opened the BBCF again so i could check the input history, it was working for a few seconds, before going crazy again after i pressed L2

i have tried updating drivers and switching HID device modes, none of those fixed this
![1640325405](https://user-images.githubusercontent.com/50655701/147323528-b4b2dcaf-b731-4e4d-a936-40cc766aa88b.png)

**Comment by nefarius** [comment]
**TL;DR:** the driver doesn't just magically "invent" random inputs out of nowhere, it takes what comes from the hardware, so conclusion: your controller is faulty in some way shape or form. I've explained that many many times in the past, driver software doesn't just come up with changing data because it feels like it 😉 

Cheers

**Reply by ZxSol** [reply]
I have 3 other ps3 controllers I have tested, one of them I just got fairly recently, all official ps3 controllers, and the issue still was persisting. All of them also work fine on a ps3 console, so its not the controller.

**Reply by nefarius** [reply]
The driver translates what it receives so... 🤷

**Comment by Kanuan** [comment]
@ZxSol Which HID Device Mode are you using?

**Comment by fraf** [comment]
Check this !
https://github.com/ViGEm/DsHidMini/discussions/221

**Reply by nefarius** [reply]
👏 It's never been the software, why do people believe that software would suddenly decide based on some randomness to break the translation logic that hasn't been touched in years? 😁

**Reply by F8RDev** [reply]
I believe every software may have bugs, I not saying these random behavior caused by DsHidMini.

But to me, you sound like saying this software is immune to bugs, and that must be a hardware problem.
And bugs has been found on many software that worked fine for years.

**Reply by nefarius** [reply]
No. Not at all what I said. I got years of experience with this stuff now and not a single case came across my desk where suddenly, randomly, magically a driver decided to just start mangling controller inputs on one day. Ofc. Software has bugs but a bug where it suddenly becomes self aware and starts rewriting its own logic? That's just silly, makes no sense and I don't waste my time investigating these cases because, to my experience, it has **never** been the softwares fault. Period.

**Comment by DaveBalda** [comment]
I had the same problem with this software. I thought it was an hardware issue since my pad was pretty dirty so I deep cleaned it (motherboard and other stuff), added some thickness to the spongy part and tried it. It worked perfectly on my console (PS3) but on the PC it had the same problem. So I uninstalled dshidmini and I replaced it with SCPToolkit and the problem was solved!
